src/Neo/SmartContract/ApplicationEngine.Runtime.cs

```diff
@@ -373,7 +373,7 @@ protected internal void RuntimeNotifyV1(byte[] eventName, Array state)
                 throw new InvalidOperationException("Notifications are not allowed in dynamic scripts.");
             using MemoryStream ms = new(MaxNotificationSize);
             using BinaryWriter writer = new(ms, Utility.StrictUTF8, true);
-            BinarySerializer.Serialize(writer, state, MaxNotificationSize);
+            BinarySerializer.Serialize(writer, state, MaxNotificationSize, Limits.MaxStackSize);
```

Comment:
It may badly affect other mainnet transactions, don't we want to create a hardfork for this change and enable limited serialization starting from some future height in mainnet?

Reply:
Is not possible to call with a state with more items than the current limit, it doesn't come from a json like the manifest